lib/puppet_x/templates/init_ps.ps1.erb

```diff
@@ -392,6 +396,12 @@ function Invoke-PowerShellUserCode
     [Void]$ps.AddScript($global:ourFunctions)
     $ps.Invoke()
 
+    if ([string]::IsNullOrEmpty($WorkingDirectory)) {
+      $ps.Runspace.SessionStateProxy.Path.SetLocation($global:DefaultWorkingDirectory)    
+    } else {
+      if (-not (Test-Path -Path $WorkingDirectory)) { Throw "Working directory `"$WorkingDirectory`" does not exist" }
+      $ps.Runspace.SessionStateProxy.Path.SetLocation($WorkingDirectory)
```

Comment:
Looks like this is ok

So a quick test:

* Created `C:\a\b\c`
* Set Deny Everyone All perms on `C:\a\b` only (This Folder and files only)

* Can `Set-Location` to `C:\a`,  `C:\a\b` and  `C:\a\b\c`
* Can only do `Get-ChildItem`  on  `C:\a` and  `C:\a\b\c`.
* `Get Access Denied` on`C:\a\b` 
* Can `Test-Path` to `C:\a`,  `C:\a\b` and  `C:\a\b\c` all with `$true`

Side note:
CMD.exe does have that problem.  Can't `cd c:\a\b\` - get's access denied.